### DISCUSSION TITLE
Restart and Recover

### DISCUSSION THREAD

**Discussion by wowodejiajia** [idaholab/moose#26542] [Q&A General] [answered]
Dear MOOSE experts,
I would like to use the result file from the previous task as the initial condition, but it won't run. Could you please help me check if there are any issues with my input file?
The original input file
```
[Mesh]
  type = GeneratedMesh
  dim = 2
  nx = 200
  ny = 200
  nz = 0
  xmin = 0
  xmax = 200
  ymin = 0
  ymax = 200
  zmin = 0
  zmax = 0
[]

[Variables]
  [./w]
  [../]
  [./eta]
  [../]
  [./pot]
  [../]
[]

[Functions]
  [./ic_func_eta]
    type = ParsedFunction
    expression = 0.5*(1.0-1.0*tanh((x-20)*2))
  [../]
  [./ic_func_c]
    type = ParsedFunction
    expression = 0
  [../]
  [./ic_func_pot]
    type = ParsedFunction
    expression = -0.225*(1.0-tanh((x-20)*2))
  [../]
[]

[ICs]
  [./eta]
    type = FunctionIC
    variable = eta
    function = ic_func_eta
  [../]
  [./w]
    type = FunctionIC
    variable = w
    function = ic_func_c
  [../]
  [./pot]
    type = FunctionIC
    variable = pot
    function = ic_func_pot
  [../]
[]
```
Input file for continuing calculation

```
[Mesh]
  file = c01_out.e
[]

[Variables]
  [./w]
    family = LAGRANGE
    order = FIRST
    initial_from_file_var = w
    initial_from_file_timestep = 245
  [../]
  [./eta]
    family = LAGRANGE
    order = FIRST
    initial_from_file_var = eta
    initial_from_file_timestep = 245
  [../]
  [./pot]
    family = LAGRANGE
    order = FIRST
    initial_from_file_var = pot
    initial_from_file_timestep = 245
  [../]
[]
```
Thank you.

**Comment by GiudGiud** [comment]
HelloI see two initialization methods here, ICs and the initial-from-file.You can only use one

**Reply by wowodejiajia** [reply]
First, I calculated c01_out.e using the first input file. Then, I used c01_out.e as the initial condition and continued the calculation with the second input file. But it didn't even finish the first step and stopped running.
I plan to repeat this step many times.

**Comment by GiudGiud** [comment]
Nevermind that 's not the problem.

In the second simulation, use a FileMeshGenerator in the `[Mesh]` block and set the `use_for_exodus_restart` parameter to true

**Reply by wowodejiajia** [reply]
I tried, but it still doesn't work.

the second input file
```
[Mesh]
  [./fmg]
    type = FileMeshGenerator
    file = 'c01_out.e'
    use_for_exodus_restart = true
  [../]
[]

[Variables]
  [./w]
    family = LAGRANGE
    order = FIRST
    initial_from_file_var = w
    initial_from_file_timestep = 245
  [../]
  [./eta]
    family = LAGRANGE
    order = FIRST
    initial_from_file_var = eta
    initial_from_file_timestep = 245
  [../]
  [./pot]
    family = LAGRANGE
    order = FIRST
    initial_from_file_var = pot
    initial_from_file_timestep = 245
  [../]
[]
```

**Reply by GiudGiud** [reply]
hello

what error do you get?
Does time step 245 exist in the file? Do you get a better result with LATEST instead of 245?

**Reply by wowodejiajia** [reply]
Time Step 0, time = 0

Time Step 1, time = 0.02, dt = 0.02
Nonlinear solve did not converge due to DIVERGED_MAX_IT iterations 20
Currently Executing
  Finished Solving                                                                       [ 34.41 s] [  178 MB]
 Solve Did NOT Converge!
Aborting as solve did not converge

Time Step 1, time = 0.01, dt = 0.01
Nonlinear solve did not converge due to DIVERGED_MAX_IT iterations 20
  Finished Solving                                                                       [ 28.42 s] [  182 MB]
 Solve Did NOT Converge!
Aborting as solve did not converge

Time Step 1, time = 0.005, dt = 0.005
Nonlinear solve did not converge due to DIVERGED_LINE_SEARCH iterations 16
  Finished Solving                                                                       [ 24.37 s] [  182 MB]
 Solve Did NOT Converge!
Aborting as solve did not converge

Time Step 1, time = 0.0025, dt = 0.0025

Stuck here. It neither reports an error nor runs here.

**Comment by wowodejiajia** [comment]
The first input file
[first.docx](https://github.com/idaholab/moose/files/13930178/first.docx)

The second input file
[second.docx](https://github.com/idaholab/moose/files/13930179/second.docx)

The article I am referring to is
[2023-ACS AMI.pdf](https://github.com/idaholab/moose/files/13930185/2023-ACS.AMI.pdf)

**Reply by GiudGiud** [reply, marked as answer]
I think some of your materials have state.

Exodus restart does not restart those. So maybe that brings your simulation in an invalid configuration upon restart (first input works fine right?)

To restart materials with state, you need to use the checkpoint restart. See the restart recover documentation page to see how to do that

**Reply by wowodejiajia** [reply]
My first input file is running normally.
I am running the input file with added checkpoints restart. How to restart after running is completed? Do you have any examples?

**Reply by GiudGiud** [reply]
There are examples of checkpoint restart in moose/test/tests/restart.
This one too :  restart in moose/test/tests/time_steppers/timesequence_stepper/timesequence_restart1.i

**Reply by wowodejiajia** [reply]
There's a problem.
Time Step 0, time = 112.721
WARNING! There are options you set that were not used!
WARNING! could be spelling mistake, etc!
There is one unused database option. It is:
Option left: name:-i value: f02.i source: command line

f02.i
[Mesh]
  file = c02_checkpoint_cp/6750-mesh.cpr
[]

[Problem]
  restart_file_base = c02_checkpoint_cp/6750
  allow_initial_conditions_with_restart = true
[]

**Reply by GiudGiud** [reply]
Hello

This is just a warning you can ignore it. 


Why do you have allow_initial_conditions during restart set? Whichever initial condition you have set overrides the restarted values.

So it s ok but only if the restarted and initialized variables are diffeeent

**Reply by wowodejiajia** [reply]
I made a mistake in end_time. Now it can run.
Thank you very much for your patience.
